### PR TITLE
Intel MKL-optimized HPL

### PR DESCRIPTION
**Pull request by dodecatheon**
Uses provided binary from intel-oneapi-mkl spack package

calculator workload modified to correctly calculate N based on 80% of available memory, with P and Q based on 2 procs per node.

To change memory usage, e.g. for testing, set percent_mem for workload calculator in ramble.yaml.

### CODE REVIEW

**Comment by dodecatheon** [comment]
Default `calculator` memory usage set to 85% of memory.

File: var/ramble/repos/builtin/applications/hpl/application.py [hunk 1]
```diff
@@ -17,120 +17,180 @@ class Hpl(SpackApplication):
...
     required_package('hpl')
 
     executable('execute', 'xhpl', use_mpi=True)
 
     workload('standard', executables=['execute'])
+    workload('calculator', executables=['execute'])
 
-    workload_variable('output_file', default='HPL.out      output file name (if any)',
+    workload_variable('output_file',
+                      default='{:13} output file name (if any)'.format('HPL.out'),
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/hpl/application.py:36]: What is this change for? I see it's fairly pervasive, but it also seems unnecessary. Is there a reason why this is pushed through everywhere?

File: var/ramble/repos/builtin/applications/hpl/application.py [hunk 2]
```diff
@@ -142,55 +202,114 @@ class Hpl(SpackApplication):
...
-            self.variables['PFACTS'] = '2'
-            self.variables['N-NBMINs'] = '1'
-            self.variables['NBMINs'] = '4'
-            self.variables['N-RFACTs'] = '1'
-            self.variables['RFACTs'] = '1'
-            self.variables['N-BCASTs'] = '1'
-            self.variables['BCASTs'] = '1'
-            self.variables['N-DEPTHs'] = '1'
-            self.variables['DEPTHs'] = '1'
+            pfact = expander.expand_var('{pfact}')
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/hpl/application.py:263]: Please change these lines to:
```
pfact = expander.expand_var(expander.expansion_str('pfact'))
```

**Review by douglasjacobsen — COMMENTED** [review]
Thanks for doing this! I think my comments relate to both application files. Let me know your thoughts.

File: var/ramble/repos/builtin/applications/hpl/application.py (hunk 2 above)
Comment (dodecatheon) [reply var/ramble/repos/builtin/applications/hpl/application.py:263]: Changed in next commit.

File: var/ramble/repos/builtin/applications/hpl/application.py (hunk 1 above)
Comment (douglasjacobsen) [reply var/ramble/repos/builtin/applications/hpl/application.py:36]: Can we pull this out into a `pad_value` function, so the `'{:13}'` is not repeated in all of the workload variables?

Comment (dodecatheon) [reply var/ramble/repos/builtin/applications/hpl/application.py]: Changed and updated.

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**